lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -279,10 +286,11 @@ def add_customer_data(post, payment, options, action = '')
       def customer_fields(payment, options)
         return if options[:customer_id]
 
-        customer_address = address(options)
+        customer_address = address(options) if valid_address?(options[:billing_address])
```

Comment:
Similar to the other request, you could move the call of this method inside of the `address` or `add_address` method.

Reply:
Got it! Done!